addons/account/models/account_move.py

```diff
@@ -823,7 +823,7 @@ def _reconcile_lines(self, debit_moves, credit_moves, field):
         """
         (debit_moves + credit_moves).read([field])
         to_create = []
-        cash_basis = debit_moves and debit_moves[0].account_id.internal_type in ('receivable', 'payable') or False
+        cash_basis = debit_moves and debit_moves[0].account_id.internal_type in ('receivable', 'payable') and not debit_moves[0].invoice_id or False
```

Comment:
why only the debit_move and not the credit_move?

Reply:
does it work for both the in_invoice and the out_invoice?

it is quite weird for me.

Regards

Reply:
Being honest, I don't really know why the debit_moves and not the credit_moves, I only did the change based on the previous criteria for creating a cash basis move (debit_moves) and works for both cases, in and out invoices.